### ISSUE TITLE
False positive in `UnusedVariable`

### ISSUE THREAD

**Issue by zedxxx** [integrated-application-development/sonar-delphi#251]
Delphi IDE version: Delphi 12.1

DelphiLint version: 1.1.0

SonarDelphi version: 1.6.0

![2](https://github.com/integrated-application-development/sonar-delphi/assets/6298130/7a3b82e2-9de3-4b2a-9a07-60b6c1a9e207)

```delphi
unit Unit1;

interface

uses
  GR32,
  GR32_Polygons,
  GR32_VectorUtils;

procedure DrawThickPolyLine(ABitmap: TBitmap32; const APoints: TArrayOfFloatPoint;
  const AColor: TColor32; const AWidth: Integer);

implementation

procedure DrawThickPolyLine(ABitmap: TBitmap32; const APoints: TArrayOfFloatPoint;
  const AColor: TColor32; const AWidth: Integer);
var
  VPoly: TArrayOfFloatPoint;
begin
  VPoly := BuildPolyLine(APoints, AWidth);

  if not ABitmap.MeasuringMode then begin
    PolygonFS(ABitmap, VPoly, AColor);
  end;

  ABitmap.Changed(MakeRect(PolygonBounds(VPoly), rrOutside));
end;

end.
```

**Comment by Cirras** [comment]
Hi @zedxxx,

I can't seem to reproduce this `UnusedVariable` FP.
(See image below)

Here's what I did:
* created a new VCL project
* cloned the `Graphics32` library and added the `Source` folder to the search path in project options
* added your `Unit1` file to the project
* built the project successfully (just to make sure it could compile)
* scanned with DelphiLint (FP issue not reproduced)
  ![image](https://github.com/integrated-application-development/sonar-delphi/assets/22178573/37ab4e34-443d-497b-8a31-b666c9c8d09e)

The analyzer **requires source code** to perform type & invocation resolution, and things can go quite haywire if dependency source code isn't in the search path (or debugger source path).

One potential thing - do you only have the `Graphics32` DCUs on your search path, and not the source files?

**Comment by Cirras** [comment]
See: https://github.com/integrated-application-development/sonar-delphi/issues/250#issuecomment-2159971922
